sqlglot/optimizer/simplify.py

```diff
@@ -530,6 +555,132 @@ def simplify_concat(expression):
     return new_args[0] if len(new_args) == 1 else concat_type(expressions=new_args)
 
 
+def simplify_datetrunc(expression):
+    """Remove redundant DATE_TRUNC functions"""
+    if isinstance(expression, exp.DateTrunc) and expression.unit.name.lower() == "day":
+        return expression.this
```

Comment:
Ah, I thought we parsed into TimestampTrunc for those.

Maybe I should add support for that function too...

Reply:
Hm you could be right I didn't double-check this

Reply:
It seems that both Snowflake and Redshift are indeed parsed into TimestampTrunc, so since there's a path to fix future issues (i.e. parse `DATE_TRUNC` into `TimestampTrunc`), this is probably ok?

Reply:
I think so. I can also check the type of `this` and make sure its a date.

Reply:
I actually think we don't need this function. The other one should handle it fine. Will remove.